lib/archethic/contracts/interpreter/condition_validator.ex

```diff
@@ -13,9 +13,38 @@ defmodule Archethic.Contracts.Interpreter.ConditionValidator do
   @doc """
   Determines if the conditions of a contract are valid from the given constants
   """
-  @spec execute_condition(ConditionsSubjects.t(), map()) ::
+  @spec execute_condition(Macro.t() | ConditionsSubjects.t(), map()) ::
           {:ok, list(String.t())} | {:error, String.t(), list(String.t())}
-  def execute_condition(conditions, constants = %{}) do
+  def execute_condition(condition_or_ast, constants = %{}) do
```

Comment:
Suggestion: Maybe you can use pattern matching to simplify the body of this function, with a one function with `%ConditionsSubjects{}` and the other one with the Macro